bpf/aya-bpf/src/maps/array.rs

```diff
@@ -47,14 +47,27 @@ impl<T> Array<T> {
         }
     }
 
+    #[inline(always)]
     pub fn get(&self, index: u32) -> Option<&T> {
-        unsafe {
-            let value = bpf_map_lookup_elem(
-                self.def.get() as *mut _,
-                &index as *const _ as *const c_void,
-            );
-            // FIXME: alignment
-            NonNull::new(value as *mut T).map(|p| p.as_ref())
-        }
+        unsafe { self.lookup(index).map(|p| p.as_ref()) }
+    }
+
+    #[inline(always)]
+    pub fn get_ptr(&self, index: u32) -> Option<*const T> {
+        unsafe { self.lookup(index).map(|p| p.as_ptr() as *const T) }
+    }
+
+    #[inline(always)]
+    pub fn get_ptr_mut(&self, index: u32) -> Option<*mut T> {
+        unsafe { self.lookup(index).map(|p| p.as_ptr() as *mut T) }
```

Comment:
as_mut_ptr()

Reply:
```
error[E0599]: no method named `as_mut_ptr` found for struct `NonNull<T>` in the current scope
  --> bpf/aya-bpf/src/maps/array.rs:62:47
   |
62 |         unsafe { self.lookup(index).map(|p| p.as_mut_ptr()) }
   |                                               ^^^^^^^^^^ help: there is a method with a similar name: `as_mut`
```

`as_mut` is not suitable for us, as it returns a reference

https://doc.rust-lang.org/core/ptr/struct.NonNull.html#method.as_mut